File: python/dp-e14.py
from itertools import chain

def split_in_chunks(ls, size):
    newls = []
    for i in xrange(0, len(ls), size):
        newls.append(ls[i:i+size])
    return newls

def reverse_chunks(chunks):
    return [x[::-1] for x in chunks]

def flatten_chunks(chunks):
    return list(chain(*chunks))

chunks = split_in_chunks(raw_input("list > ").split(), int(raw_input("chunk > ")))
reversedchunks = reverse_chunks(chunks)
flattenedchunks = flatten_chunks(reversedchunks)

print(' '.join(flattenedchunks))

File: javascript/dp-e14.js
const args = process.argv.slice[2];
const arr = args[0];
const size = args[1];

function chunks(arr, size) {
  let newArr = [];
  for (var i = 0; i < arr.length; i += size) {
    newArr.push(arr.slice(i, i+size));
  }
  return newArr;
}

function reversedChunks(arr, size) {
  return chunk(arr,size).reverse();
}

function flattenedChunks(chunks) {
  let flat = [];
  for (var i = 0; i < chunks.length; i++) {
    for (var k = 0; k < chunks[i].length; k++) {
      flat.push(chunks[i][k]);
    }
  }
  return flat;
}

console.log(flattenedChunks(reversedChunks(arr, size)).join(" "));
